consensus/blake3pow/sealer.go

```diff
@@ -0,0 +1,423 @@
+			return
+		}
+	}
+}
+
+// makeWork creates a work package for external miner.
+//
+// The work package consists of 3 strings:
+//
+//	result[0], 32 bytes hex encoded current header pow-hash
+//	result[1], 32 bytes hex encoded seed hash used for DAG
+//	result[2], 32 bytes hex encoded boundary condition ("target"), 2^256/difficulty
+//	result[3], hex encoded header number
+func (s *remoteSealer) makeWork(header *types.Header) {
+	hash := header.SealHash()
+	s.currentWork[0] = hash.Hex()
+	s.currentWork[1] = hexutil.EncodeBig(header.Number())
+	s.currentWork[2] = common.BytesToHash(new(big.Int).Div(big2e256, header.Difficulty()).Bytes()).Hex()
+
+	// Trace the seal work fetched by remote sealer.
+	s.currentHeader = header
+	s.works[hash] = header
+}
+
+// notifyWork notifies all the specified mining endpoints of the availability of
+// new work to be processed.
+func (s *remoteSealer) notifyWork() {
+	work := s.currentWork
+
+	// Encode the JSON payload of the notification. When NotifyFull is set,
+	// this is the complete block header, otherwise it is a JSON array.
+	var blob []byte
+	if s.blake3pow.config.NotifyFull {
+		blob, _ = json.Marshal(s.currentHeader)
+	} else {
+		blob, _ = json.Marshal(work)
+	}
+
+	s.reqWG.Add(len(s.notifyURLs))
+	for _, url := range s.notifyURLs {
+		go s.sendNotification(s.notifyCtx, url, blob, work)
+	}
+}
+
+func (s *remoteSealer) sendNotification(ctx context.Context, url string, json []byte, work [4]string) {
+	defer s.reqWG.Done()
+
+	req, err := http.NewRequest("POST", url, bytes.NewReader(json))
+	if err != nil {
+		s.blake3pow.config.Log.Warn("Can't create remote miner notification", "err", err)
+		return
+	}
+	ctx, cancel := context.WithTimeout(ctx, remoteSealerTimeout)
+	defer cancel()
+	req = req.WithContext(ctx)
+	req.Header.Set("Content-Type", "application/json")
+
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		s.blake3pow.config.Log.Warn("Failed to notify remote miner", "err", err)
+	} else {
+		s.blake3pow.config.Log.Trace("Notified remote miner", "miner", url, "hash", work[0], "target", work[2])
+		resp.Body.Close()
+	}
+}
+
+// submitWork verifies the submitted pow solution, returning
+// whether the solution was accepted or not (not can be both a bad pow as well as
+// any other error, like no pending work or stale mining result).
+func (s *remoteSealer) submitWork(nonce types.BlockNonce, sealhash common.Hash) bool {
+	if s.currentHeader == nil {
+		s.blake3pow.config.Log.Error("Pending work without block", "sealhash", sealhash)
+		return false
+	}
+	// Make sure the work submitted is present
+	header := s.works[sealhash]
+	if header == nil {
+		s.blake3pow.config.Log.Warn("Work submitted but none pending", "sealhash", sealhash, "curnumber", s.currentHeader.NumberU64())
+		return false
+	}
+	// Verify the correctness of submitted result.
+	header.SetNonce(nonce)
+
+	start := time.Now()
+	if !s.noverify {
+		if err := s.blake3pow.verifySeal(header); err != nil {
```

Comment:
This is different from progpow. Progpow panics and says remote verifier not supported. We need to align the two.

Reply:
removed it, already verifying seal in calcorder